geowebcache/rest/src/main/java/org/geowebcache/rest/service/SeedService.java

```diff
@@ -94,27 +91,52 @@ public ResponseEntity<?> getRunningLayerTasks(HttpServletRequest request, String
         try {
             XStream xs = new GeoWebCacheXStream(new JsonHierarchicalStreamDriver());
             long[][] list;
-            if (null == layer) {
-                list = seeder.getStatusList();
-            } else {
-                try {
-                    seeder.findTileLayer(layer);
-                } catch (GeoWebCacheException e) {
-                    HttpHeaders headers = new HttpHeaders();
-                    headers.setContentType(MediaType.TEXT_PLAIN);
-                    return new ResponseEntity<>(e.getMessage(), headers, HttpStatus.BAD_REQUEST);
-                }
-                list = seeder.getStatusList(layer);
+            try {
+                list = findTileLayer(layer);
+            } catch (GeoWebCacheException e) {
+                return getBadRequestResponse(
+                        MediaType.TEXT_PLAIN, e.getMessage(), HttpStatus.BAD_REQUEST);
             }
             JSONObject obj = new JSONObject(xs.toXML(list));
-            HttpHeaders headers = new HttpHeaders();
-            headers.setContentType(MediaType.APPLICATION_JSON);
-            return new ResponseEntity<>(obj.toString(), headers, HttpStatus.OK);
+            return getBadRequestResponse(MediaType.APPLICATION_JSON, obj.toString(), HttpStatus.OK);
         } catch (JSONException jse) {
             LOG.log(Level.SEVERE, jse.getMessage(), jse);
-            HttpHeaders headers = new HttpHeaders();
-            headers.setContentType(MediaType.TEXT_PLAIN);
-            return new ResponseEntity<Object>("error", headers, HttpStatus.INTERNAL_SERVER_ERROR);
+            return getBadRequestResponse(
+                    MediaType.TEXT_PLAIN, "error", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    public ResponseEntity<?> getRunningLayerTasksXml(HttpServletRequest request, String layer) {
+        try {
+            XStream xs = new GeoWebCacheXStream();
+            long[][] list;
+            try {
+                list = findTileLayer(layer);
+            } catch (GeoWebCacheException e) {
+                return getBadRequestResponse(
+                        MediaType.TEXT_PLAIN, e.getMessage(), HttpStatus.BAD_REQUEST);
+            }
+            return getBadRequestResponse(MediaType.APPLICATION_XML, xs.toXML(list), HttpStatus.OK);
+        } catch (JSONException jse) {
+            LOG.log(Level.SEVERE, jse.getMessage(), jse);
+            return getBadRequestResponse(
+                    MediaType.TEXT_PLAIN, "error", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    private ResponseEntity<?> getBadRequestResponse(
+            MediaType textPlain, String message, HttpStatus badRequest) {
```

Comment:
```suggestion
    private ResponseEntity<?> getResponse(
            MediaType mediaType, String message, HttpStatus status) {
```

Make method and parameter names more generic, at a first sight this method seems to be designed to return an error, but in fact it can return any response whose body is textual.

Reply:
Done